Applications/Spire/Source/Ui/IconButton.cpp

```diff
@@ -39,44 +45,31 @@ IconButton::IconButton(QImage icon, Style style, QWidget* parent)
     })").arg(scale_height(10)).arg(scale_height(2)).arg(scale_width(6)));
 }
 
-connection IconButton::connect_clicked_signal(
-    const ClickedSignal::slot_type& slot) const {
-  return m_clicked_signal.connect(slot);
-}
-
-void IconButton::focusInEvent(QFocusEvent* event) {
-  m_last_focus_reason = event->reason();
-  update();
-  QWidget::focusInEvent(event);
-}
-
-void IconButton::focusOutEvent(QFocusEvent* event) {
-  update();
-  QWidget::focusOutEvent(event);
-}
-
 void IconButton::keyPressEvent(QKeyEvent* event) {
-  if(event->key() == Qt::Key_Enter || event->key() == Qt::Key_Return ||
-      event->key() == Qt::Key_Space) {
-    m_clicked_signal();
+  if(!event->isAutoRepeat() && (event->key() == Qt::Key_Enter ||
+      event->key() == Qt::Key_Return)) {
     event->accept();
+    setDown(true);
+    Q_EMIT pressed();
     return;
   }
-  event->ignore();
+  QAbstractButton::keyPressEvent(event);
 }
 
-void IconButton::mousePressEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton) {
+void IconButton::keyReleaseEvent(QKeyEvent* event) {
+  if(!event->isAutoRepeat() && isDown()) {
     event->accept();
+    setDown(false);
+    Q_EMIT released();
+    Q_EMIT clicked(isChecked());
+    return;
   }
+  QAbstractButton::keyReleaseEvent(event);
 }
 
-void IconButton::mouseReleaseEvent(QMouseEvent* event) {
-  if(event->button() == Qt::LeftButton) {
-    if(rect().contains(event->localPos().toPoint())) {
-      event->accept();
-      m_clicked_signal();
-    }
+void IconButton::mouseMoveEvent(QMouseEvent* event) {
+  if(rect().contains(event->pos())) {
+    QAbstractButton::mouseMoveEvent(event);
```

Comment:
What does this do?

Reply:
Previously, when I was using the released signal it would be emitted if the button was clicked and held, then the cursor was moved away from the button. I wanted to prevent the released signal from triggering here, but because the clicked signal more closely represents the behavior we want this is no longer needed.

This would have made the behavior inconsistent, anyway.